# Ujian Akhir Semester (5)

Nama   : Clarence Chandra Sentosa

NRP    : 1973010

# Training Classifier dengan Random Forest

 **Random Forest** termasuk model machine learning yang umum ditemui dan merupakan pengembangan dari model **Desicion Tree**.

# Load Breast Cancer Dataset

In [3]:
# Code 01

from sklearn.datasets import load_breast_cancer

X, y = load_breast_cancer(return_X_y=True)

print(f'Dimensi Feature: {X.shape}')
print(f'Class: {set(y)}')

Dimensi Feature: (569, 30)
Class: {0, 1}


**Konsep berpikir code 01** :
- Pertama-tama kita akan import terlebih dahulu modulnya dengan memanggil method *'from sklearn.datasets import load_breast_cancer'*.
- Lalu berikutnya kta akan memanggil *'load_breast_cancer'* dengan menyertakan parameter *'(return_X_y)'* yang kita beri nilai *True*. Konsekuensinya disini kita harus menyiapkan 2 buah variabel yaitu *X* dan *y*, dimana *X* akan menampung sekumpulan nilai feature dan *y* akan menampung sekumpulan nilai target labelnya. 
- Lalu selanjutnya kita akan coba print out atau tampilkan shape atau dimensi dari variabel *X* nya, dan juga akan kita coba print out atau tampilkan class apa saja yang terdapat pada target label *y* nya.

Jika kita perhatikan pada output code 01, dimensi feature atau dimensi dari variabel *X* nya terdiri dari 569 baris dan 30 kolom. Lalu terkait target label *y* nya, disini terdapat 2 buah class atau 2 nilai class yang berbeda yaitu class 0 dan class 1.

# Training Set dan Testing Set

Selanjutnya dataset akan dibagi kedalam *training set* dan *testing set*.

In [4]:
# Code 02

from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X,
                                                    y,
                                                    test_size=0.4,
                                                    random_state=0)

**Konsep berpikir code 02** :
- Disini pertama-tama kita akn import modul terlebih dahulu dengan memanggi method *'from sklearn.model_selection import train_test_split'*. 
- Lalu berikutnya kita akan panggil *'train_test_split'* dengan menyertakan beberapa parameter yaitu *'(X, y)'*.
- Lalu kita juga akan menentukan *'(test_size)'* sebesar 0.4. Artinya ukuran *testing_set* nya adalah 40%, sedangkan 60% nya akan kita alokasikan sebagai *Training Set*. 
- Selanjutnya untuk *'(random_state)'* nya akan kita pilih atau set sebagai 0. 
- Berikutnya pemanggilan *train_test_split* ini akan membutuhkan 4 buah variabel. Oleh karenanya, pada kasus ini kita akan siapkan *X_train, X_test, y_train,* serta *y_test* sebagai variabelnya.

Disini kita sudah memiliki feature untuk Training set *X_Train* dan juga feature untuk testing set *X_test*, lalu kita juga memiliki target label untuk yraining set *y_train*, dan juga target label untuk testing setnya *y_test*.

## Classification dengan `RandomForestClassifier`

Disini kita akan menerapkan **Random Forest** untuk melakukan klasifkasi **Breast Cancer**.

In [6]:
# Code 03

from sklearn.ensemble import RandomForestClassifier

model = RandomForestClassifier(n_estimators=150, 
                               random_state=0)

model.fit(X_train, y_train)

RandomForestClassifier(n_estimators=150, random_state=0)

**Konsep berpikir code 03** :
- Pertama-tama kita akan mengimport terlebihi dahulu modulnya dengan memanggil method *'from sklearn.ensemble import RandomForestClassifier'*.
- Selanjutnya kita akan bentuk objek dari *'RandomForestClassifier'* ini dengan menyertakan 2 buah parameter yaitu *'(n_estimators=150)'* yang untuk kasus kali ini kita melakukan training 150 buah model.
- Lalu agar eksperimen kita ini bisa kita replicate dengan hasil yang konsisten, kita juga perlu mengeset random statenya yang dalam kasus kita kali ini adalah *'(random_state=0)'*. 
- Selanjutnya objek yang terbentuk tersbut akan kita tampung kedalam variabel *model*. 
- Untuk selanjutnya, model ini akan kita training dengan memanggil method *fit* dengan menyertakan parameter *'(X_train, y_train)'*.

# Evaluasi Model

Disini kita akan lakukan evaluasi performa dari model **Random Forest** yang baru saja kita training tadi dengan memanfaatkan **Classification Report**.

In [7]:
# Code 04

from sklearn.metrics import classification_report

y_pred = model.predict(X_test)

print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       0.96      0.87      0.91        83
           1       0.93      0.98      0.95       145

    accuracy                           0.94       228
   macro avg       0.94      0.92      0.93       228
weighted avg       0.94      0.94      0.94       228



**Alur berpikir code 04** :
- Pertama-tam kita akan import terlebih dahulu modulnya dengan memanggil *'from sklearn.metrics import classification_report'*.
- Lalu berikutnya kita lakukan prediksi terhadap x_test dengan memanggil *'model.predict(X_test)'* yang akan kita tampung kedalam variabel *y_pred*.
- Untuk selanjutnya kita akan memanggil *'classification_report'* dengan menyertakan parameter *'(y_test, y_pred)'*.

Hasil output code 04 merupakan *classification report* sebagai bentuk evaluasi terhadap model **Random Forest** yang baru saja kita training sebelumnya, ditampilkan nilai *precision*, *recall*, serta *f1-score* nya.

Demikianlah cara yang saya lakukan untuk melakukan training classifier dengan menggunakan *Random Forest* pada **Breast Cancer** dataset.

# "Banyak Belajar Biar Bisa Bantu Banyak Orang"

Written by : Clarence Code Pianist (1973010)